components/script/dom/bindings/utils.rs

```diff
@@ -48,6 +49,41 @@ use crate::dom::bindings::trace::trace_object;
 use crate::dom::windowproxy::WindowProxyHandler;
 use crate::script_runtime::JSContext as SafeJSContext;
 
+/// A transparent wrapper intended for use with types that are not considered threadsafe by
+/// the Rust compiler but need to be used in contexts that require threadsafety. This is
+/// needed to allow using JS API types (which usually involve raw pointers) in static initializers,
+/// when Servo guarantees through the use of OnceLock that only one thread will ever initialize
+/// the value.
+struct ForceThreadSafe<T>(pub T);
+
+/// A OnceLock wrapping a type that is not considered type-safe by the Rust compiler, but can
+/// will be used in a type-safe manner.
+pub struct ThreadUnsafeOnceLock<T>(OnceLock<ForceThreadSafe<T>>);
```

Comment:
Actually initializing multiple times it's not a reason why we need `ForceThreadSafe` (this is prevented by OnceLock anyway). If we would have any interior mutable type inside T that we would mutate that would be unsafe. So technically only ThreadUnsafeOnceLock.get is unsafe (must not mutate interior). We know that's currently not happening because statics were const before which forbids interior mut.

In the future we could use https://doc.rust-lang.org/nightly/std/marker/trait.Freeze.html to ensure that (or write crown lint), but for know we should just mark ThreadUnsafeOnceLock.get as unsafe and document this better to prevent it becoming a footgun in the future.